src/main/java/org/aszjch/demoapp/infrastructure/storage/MultipartFileToFileConverter.java

```diff
@@ -0,0 +1,43 @@
+package org.aszjch.demoapp.infrastructure.storage;
+
+import jakarta.annotation.PostConstruct;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.aszjch.demoapp.infrastructure.config.TempFilesProperties;
+import org.springframework.stereotype.Component;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class MultipartFileToFileConverter {
+
+    private final TempFilesProperties tempFilesProperties;
+
+    @PostConstruct
+    public void init() {
+        final Path path = Path.of(tempFilesProperties.getDir());
+        final File pathAsFile = path.toFile();
+        if (!pathAsFile.exists() || !pathAsFile.isDirectory()) {
+            final boolean result = pathAsFile.mkdir();
+            log.debug("Temp path creation result: {}", result);
+        }
+    }
+
+    public File convertMultipartFile(final MultipartFile multipartFile) throws IOException {
```

Comment:
Zastanawiam się czy tych bajerów nie da się upchnąć w naszej klasie CustomMultiPartFile ale na razie oglądam to na githubie to ciężko powiedzieć.

Reply:
Wszystko zależy jaki MultipartFile nam w łapki wpadnie. Jak ten nasz customowy, to tak naprawdę wystarczyłby nam getFile na nim. Natomiast jeżeli wpadnie faktyczny MultipartFile springowy, to już trochę ciężko będzie...